packages/hardhat-core/src/internal/util/download.ts

```diff
@@ -27,47 +21,48 @@ export async function download(
   timeoutMillis = 10000
 ) {
   const { pipeline } = await import("stream");
-  const { default: fetch } = await import("node-fetch");
+  const { getGlobalDispatcher, ProxyAgent, request } = await import("undici");
   const streamPipeline = util.promisify(pipeline);
-  const fetchOptions: FetchOptions = {
-    timeout: timeoutMillis,
-    agent: undefined,
-  };
 
-  // Check if Proxy is set https
-  if (process.env.HTTPS_PROXY !== undefined) {
-    // Create the proxy from the environment variables
-    const proxy: string = process.env.HTTPS_PROXY;
-    fetchOptions.agent = new HttpsProxyAgent.HttpsProxyAgent(proxy);
-  }
+  const dispatcher = (() => {
```

Comment:
I don't understand this pattern

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2370/commits/fd1528fdcf62a14950baf21206c21f3845133d8d